[33기 김혜수] ADD : Signin/Signup Layout

keemhyesu opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [X] 기능 추가
- [X] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
1. Signin, Signup 레이아웃(html,css) 구현
2. Signin, Signup 유효성 검사 후 로그인,회원가입 버튼 활성화 기능
(아이디-'@', '.' 포함/비밀번호-숫자,문자,특수문자 1자 이상 포함 총 8자 이상/약관 체크박스: (필수)2가지 또는 전체 선택 시 버튼 활성화)

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
<레이아웃 부분>
1. display: flex, justify-content: center; flex-direction: column; 속성을 이용하여 전체적인 큰 틀을 잡았습니다. 
2. Signin 페이지에서 input태그의  type: checkbox; 속성을 처음 사용했습니다.
3. input 태그의 focus outline: none; 속성을 사용했습니다.

<기능 부분>
1. 아이디, 비밀번호 유효성 검사 부분에서 test 메소드(주어진 문자열이 정규표현식을 만족하는지 판별하고, 그 여부를 true 또는 false로 반환)를 사용하여 기능 구현을 완료했습니다.
2. 반복되는 input 부분(아이디,비밀번호,이름,비밀번호,비밀번호 확인)을 component화 하였습니다.
3. 2번 부분에서 사용할 상수 데이터를 만들고 map 메소드를 사용하였습니다.

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
<레이아웃 부분>
1. div태그를 잘 활용하기=> justify-content, flex-direction 등 div 태그로 제대로 감싸주면 레이아웃 구현은 비교적 쉽다!
2. checkbox의 크기를 조정하고 싶을 땐  zoom 속성을 이용하고 색상을 바꾸고 싶으면 acccent-color 속성을 사용하면 된다.

<기능 부분>
1. 반복되는 부분은 component화 해서 보다 깔끔하게 코드를 수정할 수 있었다.
2. map 메소드는 내 생각보다 굉장히 자주 쓰이는 것 같다..map 공부를 더 열심히 해야겠다.
3. 멘토님께서 라이브 코드 리뷰 때 약관 체크박스 부분 코드를 간결하게 하는 방법을 알려주셨는데 프로젝트 끝나고 수정을 해봐야겠다.




<br />

keemhyesu committed:
ADD: Signin html,css 완료

keemhyesu committed:
Modify : Signup layout 완료

keemhyesu committed:
Modify : 최종 layout

keemhyesu committed:
Modify : sider 오류 수정

keemhyesu committed:
Modify : sider 오류 수정

yeonuk-hwang commented on src/index.js:
```diff
@@ -3,6 +3,7 @@ import ReactDOM from 'react-dom/client';
 import Router from './Router';
 import './styles/common.scss';
 import './styles/reset.scss';
+// import Signin from './pages/Signin/Signin';
```
불필요한 주석 및 코드인 것 같네요!!

PR 올리신 후에는 반드시 github에서 File Changed 탭 확인해서 내가 작성한 코드에 실수는 없는지, 내가 의도하지 않은 작업물들이 올라가지 않았는지 확인하는 습관을 들여주세요!

yeonuk-hwang commented on src/pages/Signin/Signin.js:
```diff
@@ -0,0 +1,85 @@
+  //   navigate('/main-hyesu');
+  // };
+
+  // const disable = inputId.includes('@') && inputPw.length >= 1 ? true : false;
```
불필요한 주석은 삭제해주세요!
코드가 주석처리된 채로 남아있다면 필요한 상황에 제대로 사용되거나 삭제되는 일은 거의 없고 결국 계속 남아있게 됩니다.
그러다보면 이제 작성한 사람이 아닌 다른 개발자는 필요한 코드라서 주석처리한게 아닐까? 라는 생각으로 지우지 못하고
결국 그 누구도 왜 있는지는 모르면서 계속 남아있는 불필요한 레거시가 되어버립니다! 
따라서 반드시 필요한 경우가 아니면 코드를 주석처리하기보다는 삭제해주시고 만약 다시 보고싶다면 커밋 이력에 잘 남겨뒀다가 원하는 시점에 다시 보면 됩니다!

yeonuk-hwang replied:
사용할려는 기능이면, 코드 작성해주시면 되고 아니라면 삭제헤주시면 될 것 같네요!

이전 코드 참고하는 것은 좋지만 그대로 카피 페이스트 하는 것 보다 필요한 부분들을 잘 조합해서 수정해서 사용하시는게 좋습니다!!

yeonuk-hwang commented on src/pages/Signin/Signin.js:
```diff
@@ -0,0 +1,85 @@
+  // const disable = inputId.includes('@') && inputPw.length >= 1 ? true : false;
+
+  return (
+    <div>
```
전체적으로 최상위 태그의 className을 컴포넌트 이름과 동일하게 부여한 뒤 그 className하위로 네스팅 하는 부분이 적용안되어있습니다. 이렇게 되면 동일한 className을 가진 다른 컴포넌트의 요소들에게도 영향을 미칠 수 있습니다.
**최상위 태그의 className을 컴포넌트 이름과 동일하게 부여한 뒤 그 className하위로 네스팅** 적용해주세요

예시)

```jsx
const ProductDetail = () => {

	return(
		<section className="productDetail">
			~~~~
		</section>
	)
}
```

yeonuk-hwang commented on src/pages/Signin/Signin.scss:
```diff
@@ -0,0 +1,119 @@
+.loginForm {
```
네스팅 적용해주세요!!

yeonuk-hwang commented on src/pages/Signin/Signin.scss:
```diff
@@ -0,0 +1,119 @@
+  border: 0.063rem solid #fabf01;
+  border-radius: 0.313rem;
+  margin-bottom: 1.875rem;
+}
```
네스팅 잘 되어있다고 하면  tag.className 형태로 안쓰고 그냥 .className으로만 작성해도 될 것 같네요!

yeonuk-hwang commented on src/pages/Signup/Signup.scss:
```diff
@@ -0,0 +1,149 @@
+  right: 1.25rem;
+  margin-top: 1.25rem;
+
+  .terms {
+    margin-bottom: 0.625rem;
+  }
+  .agreeTerms {
+    margin-bottom: 0.625rem;
+  }
+  .agreeTerms2 {
+    margin-bottom: 0.625rem;
+  }
+  .agreeTerms3 {
+    margin-bottom: 0.625rem;
+  }
```
```suggestion
  .terms {
    margin-bottom: 0.625rem;
  }
```
동일한 스타일을 부여할 것이라면, 각자 다른 className을 줄 필요 없이 하나의 className으로 처리하면 될 것 같네요!!

className의 궁극적인 목적은 스타일을 주기 위함입니다, 항목들이 다르다고 똑같은 스타일을 부여할 것인데 다른 className을 부여할 필요는 없습니다!

yeonuk-hwang replied:
다른 부분도 마찬가지로 적용되는 코멘트니까 전반적으로 확인 및 수정해주세요~!

yeonuk-hwang commented on src/pages/Signup/Signup.scss:
```diff
@@ -0,0 +1,149 @@
+  position: relative;
+  font-size: 0.875rem;
+  right: 1.25rem;
+  margin-top: 1.25rem;
```
- CSS 속성은 레이아웃에 영향을 많이 주는 순서대로, 인접 속성끼리 묶어서 
작성해주는게 좋습니다. 
- 일전에 안내해드린 [css 속성순서 컨벤션](https://study.wecode.co.kr/session/content/193) 확인하시고 지켜서 작성해주세요

yeonuk-hwang commented on src/pages/Signup/Signup.js:
```diff
@@ -0,0 +1,85 @@
+          <div className="signUpField">
+            <header>회원 가입</header>
+          </div>
+          <div className="idContent">
+            <span className="title">아이디 *</span>
+            <input
+              className="userId"
+              type="text"
+              placeholder="이메일 주소 전체를 입력해 주세요."
+            />
+          </div>
+
+          <div className="nameContent">
+            <span className="title">이름 *</span>
+            <input className="isUserName" type="text" />
+          </div>
+          <div className="passwordContent">
+            <span className="title">비밀번호 *</span>
+            <input
+              className="userPassword"
+              type="text"
+              placeholder="8자 이상 입력해 주세요."
+            />
+          </div>
+          <div className="passwordCheckContent">
+            <span className="title">비밀번호 확인 *</span>
+            <input className="userPasswordCheck" type="text" />
+          </div>
+          <div className="phoneNumber">
+            <span className="title">휴대폰 번호 *</span>
+            <input className="userPhoneNumber" type="text" />
+          </div>
```
반복되는 UI로 보이네요, 반복되는 UI는 하드코딩 하는 것 보다 Array.map 메서드를 활용하면 코드가 깔끔해지고, 유지보수에 용이해집니다.

yeonuk-hwang reviewed (changes requested):
고생하셨습니다 혜수님! 리뷰 확인 및 반영해주세요!

yeonuk-hwang commented:
혜수님 리뷰 반영하고 계시죠?? 혹시 다른 작업 하고 계신겨면 리뷰 반영 우선적으로 진행해주시고, 반영 하시고 push 및 라벨 변경해주세요!

keemhyesu committed:
Modify : 리뷰 수정 반영

yeonuk-hwang commented:
혜수님 리뷰요청 전 사이더 코멘트 해결해주세요!!, 브랜치 머지되면서 컨플릭 발생했는데 컨플릭도 해결해주세요!!
package-lock.json 컨플릭트는 아래 코멘트 확인해보시면 됩니다!

- lock 파일은 패키지 변화 히스토리 트리가 담겨있어서 프로젝트 버전이랑 함께 관리 되어야 합니다. lock 파일이 없다면 현재 프로젝트에서 사용하고 있는 버전이 아닌 최신 버전이 계속 설치됩니다.
- lock.json 파일 컨플릭트 해결 방법
    1. package.json 파일의 컨플릭트 해결
    2. `npm install --package-lock-only`
- 참고 링크 : [https://dltlabs.medium.com/package-json-vs-package-lock-json-c8d5deba12cb](https://dltlabs.medium.com/package-json-vs-package-lock-json-c8d5deba12cb)

keemhyesu commented on src/components/SignUpForm.js:
```diff
@@ -0,0 +1,41 @@
+import './SignUpForm.scss';
+
+const SignUpForm = ({ id, title, type, name, placeholder, value }) => {
+  const [inputValue, setInputValue] = useState({
```
@younuk23 연욱님 ㅠㅠㅠㅜㅠ 이부분 해결을 못하고있습니다..help me..도와주세요...

keemhyesu commented on src/components/SignUpForm.js:
```diff
@@ -0,0 +1,41 @@
+import './SignUpForm.scss';
+
+const SignUpForm = ({ id, title, type, name, placeholder, value }) => {
+  const [inputValue, setInputValue] = useState({
```
@younuk23 이 부분도...위와 마찬가지입니다..ㅜㅜㅜㅜ

keemhyesu committed:
Modify: 수정 요청 반영

keemhyesu committed:
Merge : main merge

yeonuk-hwang commented:
혜수님 사이더 해결 해주시고, 라벨 변경해주세요!

keemhyesu committed:
Modify : sider 오류 수정

keemhyesu committed:
Modify : Footer 컴포넌트 파일로 이동, 주석처리 된 부분 삭제 반영

keemhyesu reviewed (commented):
-

keemhyesu commented:
> asdasdasdasd

keemhyesu committed:
Modify : sider 오류 수정 반영

keemhyesu committed:
"Modify : sider 오류 수정 반영"

-Signin 부분의 유효성검사를 위해 정규식코드를 입력하였는데 이 부분에서
sider 오류를 감지해서 수정
-console.log와 주석처리 된 부분 삭제

keemhyesu committed:
"Modify : sider 오류 수정 반영"

-Mycart, MyCartForm 파일에서 미사용 icon 코드 삭제 완료

yeonuk-hwang commented:
- conflict 해결해주세요!, lock 파일 컨플릭트 해결은 아래 코멘트 참고해주세요
- lock 파일은 패키지 변화 히스토리 트리가 담겨있어서 프로젝트 버전이랑 함께 관리 되어야 합니다. lock 파일이 없다면 현재 프로젝트에서 사용하고 있는 버전이 아닌 최신 버전이 계속 설치됩니다.
- lock.json 파일 컨플릭트 해결 방법
    1. package.json 파일의 컨플릭트 해결
    2. `npm install --package-lock-only`
- 참고 링크 : [https://dltlabs.medium.com/package-json-vs-package-lock-json-c8d5deba12cb](https://dltlabs.medium.com/package-json-vs-package-lock-json-c8d5deba12cb)

keemhyesu committed:
Modify : conflict 오류 해결 이전  commit 진행

keemhyesu committed:
Modify: package-lock.json conflict resolve

keemhyesu committed:
Modify: confilct resolve

keemhyesu committed:
Modify : 로그인/회원가입 페이지  최종 수정 반영

yeonuk-hwang reviewed (commented):
LGTM, 리팩토링 하실 때 라이브 코드 리뷰 때 말씀드린 allCheck state 없애는 것도 꼭 시도해보세요!
